scene/main/node.cpp

```diff
@@ -1327,8 +1327,10 @@ int Node::get_child_count() const {
 }
 
 Node *Node::get_child(int p_index) const {
-	ERR_FAIL_INDEX_V(p_index, data.children.size(), nullptr);
-
+	ERR_FAIL_COND_V(p_index >= data.children.size() || p_index < -data.children.size(), nullptr);
+	if (p_index < 0) {
```

Comment:
I wonder whether this could benefit from branch prediction (#11461):

```suggestion
	if (unlikely(p_index < 0)) {
```

But I guess it wouldn't be too unlikely once people start to use negative indices. 😛

Reply:
Not really. Branch prediction has a significantly higher cost when the other case happen. So if there are cases where a negative index is used often, this could impact performances.

Reply:
Even if people start using them, it's still rather unlikely, so I'll commit. Thanks!

Reply:
No, honestly don't. 
Even if it's not used often, the code could be called several times in the frame. So it's important the performances stay correct in both cases.